internal/terraform/node_resource_abstract_instance.go

```diff
@@ -2542,7 +2589,7 @@ func (n *NodeAbstractResourceInstance) apply(
 	// where it seems like Terraform is failing to take any action at all,
 	// so we'll generate some errors to draw attention to it.
 	if !diags.HasErrors() {
-		if change.Action == plans.Delete && !newVal.IsNull() {
+		if (change.Action == plans.Delete) && !newVal.IsNull() {
```

Comment:
nit; don't need the brackets here right (and below)?

Reply:
Thanks, I previously had `|| change.Action == plans.Forget` here but removed it because the provider isn't called at all during forget.